manager/integration/tests/test_zone.py

```diff
@@ -0,0 +1,190 @@
+import pytest
+import time
+
+from common import client # NOQA
+from common import core_api  # NOQA
+from common import volume_name # NOQA
+from common import create_and_check_volume
+from common import get_self_host_id
+from common import wait_for_volume_replica_count
+from random import randrange
+from common import wait_for_volume_healthy
+from common import wait_for_volume_degraded
+from test_scheduling import wait_new_replica_ready
+from common import RETRY_COUNTS
+from common import RETRY_INTERVAL
+
+
+ZONE1 = "lh-zone1"
+ZONE2 = "lh-zone2"
+
+K8s_ZONE_LABEL = "topology.kubernetes.io/zone"
+
+
+@pytest.fixture
+def k8s_node_zone_tags(client, core_api):  # NOQA
+
+    lh_nodes = client.list_node()
+
+    node_index = 0
+    for node in lh_nodes:
+        node_name = node.name
+
+        if node_index % 2 == 0:
+            zone = ZONE1
+        else:
+            zone = ZONE2
+
+        payload = {
+            "metadata": {
+                "labels": {
+                    K8s_ZONE_LABEL: zone}
+            }
+        }
+
+        core_api.patch_node(node_name, body=payload)
+        node_index += 1
+
+    yield
+
+    lh_nodes = client.list_node()
+
+    node_index = 0
+    for node in lh_nodes:
+        node_name = node.name
+
+        payload = {
+            "metadata": {
+                "labels": {
+                    K8s_ZONE_LABEL: None}
+            }
+        }
+
+        core_api.patch_node(node_name, body=payload)
+
+
+def wait_longhorn_node_zone_updated(client, node_name): # NOQA
+    for i in range(RETRY_COUNTS):
+        lh_node = client.by_id_node(node_name)
+        if lh_node.zone != '':
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    assert lh_node.zone != ''
+
+
+def test_zone_tags(client, core_api, volume_name, k8s_node_zone_tags):  # NOQA
+
+    for lh_node in client.list_node():
+        wait_longhorn_node_zone_updated(client, lh_node.name)
+
+    volume = create_and_check_volume(client, volume_name, num_of_replicas=2)
+
+    host_id = get_self_host_id()
+
+    volume.attach(hostId=host_id)
+
+    volume = wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    zone1_replica_count = 0
```

Comment:
The lines 91-102 are redundant. The logic is the same as that of the lines 125-146.

Reply:
fixed